pandas/core/indexes/base.py

```diff
@@ -2902,16 +2895,10 @@ def union(self, other, sort=None):
                 # <T>   | <T>    -> T
                 # <T>   | <U>    -> object
                 if not (is_integer_dtype(self.dtype) and is_integer_dtype(other.dtype)):
-                    # error: Incompatible types in assignment (expression has type
-                    # "str", variable has type "Union[dtype[Any], ExtensionDtype]")
-                    dtype = "float64"  # type: ignore[assignment]
+                    dtype = np.dtype("float64")
```

Comment:
`dtype` is passed to `astype` which accepts a string? i'm not convinced this is the best approach to removing the ignores.  maybe could add a type variable declaration to `dtype = find_common_type(...` to avoid the `Incompatible types in assignment...` error.